airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationTest.java

```diff
@@ -108,106 +110,128 @@ class BigQueryDestinationTest {
-        .build()
-        .getService();
+    //all successful configs use the same project ID
+    projectId = config.get(BigQueryConsts.CONFIG_PROJECT_ID).asText();
+
+    //configWithProjectId - config that uses project:dataset notation for datasetId
+    final String dataSetWithProjectId = String.format("%s:%s", projectId, datasetId);
+    configWithProjectId = BigQueryDestinationTestUtils.createConfig(CREDENTIALS_STANDARD_INSERT_PATH, dataSetWithProjectId);
+
+    //configWithBadProjectId - config that uses "fake" project ID and should fail
+    final String dataSetWithBadProjectId = String.format("%s:%s", "fake", datasetId);
+    configWithBadProjectId = BigQueryDestinationTestUtils.createConfig(CREDENTIALS_BAD_PROJECT_PATH, dataSetWithBadProjectId);
+
+    //config that has insufficient privileges
+    insufficientRoleConfig = BigQueryDestinationTestUtils.createConfig(CREDENTIALS_NO_DATASET_CREATION_PATH, datasetId);
+    //config that tries to write to a project with disabled billing (free tier)
+    nonBillableConfig = BigQueryDestinationTestUtils.createConfig(CREDENTIALS_NON_BILLABLE_PROJECT_PATH, "testnobilling");
+    //config with GCS staging
+    gcsStagingConfig = BigQueryDestinationTestUtils.createConfig(CREDENTIALS_WITH_GCS_STAGING_PATH, datasetId);
 
-    final String datasetId = Strings.addRandomSuffix(DATASET_NAME_PREFIX, "_", 8);
     MESSAGE_USERS1.getRecord().setNamespace(datasetId);
     MESSAGE_USERS2.getRecord().setNamespace(datasetId);
     MESSAGE_TASKS1.getRecord().setNamespace(datasetId);
     MESSAGE_TASKS2.getRecord().setNamespace(datasetId);
 
     catalog = new ConfiguredAirbyteCatalog().withStreams(Lists.newArrayList(
         CatalogHelpers.createConfiguredAirbyteStream(USERS_STREAM_NAME, datasetId,
-            io.airbyte.protocol.models.Field.of("name", JsonSchemaType.STRING),
-            io.airbyte.protocol.models.Field
-                .of("id", JsonSchemaType.STRING))
+                io.airbyte.protocol.models.Field.of("name", JsonSchemaType.STRING),
+                io.airbyte.protocol.models.Field
+                    .of("id", JsonSchemaType.STRING))
             .withDestinationSyncMode(DestinationSyncMode.APPEND),
         CatalogHelpers.createConfiguredAirbyteStream(TASKS_STREAM_NAME, datasetId, Field.of("goal", JsonSchemaType.STRING))));
 
-    final DatasetInfo datasetInfo = DatasetInfo.newBuilder(datasetId).setLocation(DATASET_LOCATION).build();
-    dataset = bigquery.create(datasetInfo);
-
-    config = Jsons.jsonNode(ImmutableMap.builder()
-        .put(BigQueryConsts.CONFIG_PROJECT_ID, projectId)
-        .put(BigQueryConsts.CONFIG_CREDS, credentialsJson.toString())
-        .put(BigQueryConsts.CONFIG_DATASET_ID, datasetId)
-        .put(BigQueryConsts.CONFIG_DATASET_LOCATION, DATASET_LOCATION)
-        .put(BIG_QUERY_CLIENT_CHUNK_SIZE, 10)
-        .build());
-
-    tornDown = false;
-    addShutdownHook();
+    configs  = new HashMap<String, JsonNode>() {{
```

Comment:
Just curious if there was motivation to remove the `ImmutableMap.builder()` flow here?